install_files/ansible-base/roles/tails-config/files/securedrop_init.py

```diff
@@ -48,7 +50,27 @@
                  '/org/gnome/nautilus/preferences/automatic-decompression',
                  'false'])
 
-# notify the user
+# Set journalist.desktop and source.desktop links as trusted with Nautilus (see
+# https://github.com/freedomofpress/securedrop/issues/2586)
+# set euid and env variables to amnesia user
+os.setresgid(1000, 1000, -1)
+os.setresuid(1000, 1000, -1)
+env = os.environ.copy()
+env['XDG_RUNTIME_DIR'] = '/run/user/1000'
+env['XDG_DATA_DIR'] = '/usr/share/gnome:/usr/local/share/:/usr/share/'
+env['HOME'] = '/home/amnesia'
+env['LOGNAME'] = 'amnesia'
+env['DBUS_SESSION_BUS_ADDRESS'] = 'unix:path=/run/user/1000/bus'
+
+# remove existing shortcut, recreate symlink and change metadata attribute to trust .desktop
+for shortcut in ['source.desktop', 'journalist.desktop']:
+    subprocess.call(['rm', str(path_desktop + shortcut)], env=env)
```

Comment:
Is the `str()` necessary here? Seems like `path_desktop` and `shortcut` are both strings and concatenating them will also produce a string

(This comment applies to the next two lines)